pandas/core/groupby.py

```diff
@@ -1329,6 +1329,27 @@ def cumsum(self, axis=0):
 
     @Substitution(name='groupby')
     @Appender(_doc_template)
+    def rank(self, axis=0, method='average', numeric_only=None,
+             na_option='keep', ascending=True, pct=False):
+        """Compute numerical data ranks (1 through n) along axis.
+        """
+
+        def f(values):
+            return values.rank(axis=axis, method=method,
+                               na_option=na_option, ascending=ascending,
+                               pct=pct)
+
+        try:
+            return self.transform(f)
+        except ValueError as e:
+            obj = self._obj_with_exclusions
```

Comment:
why are you adding this????

you simply need to handle ``numeric_only`` THEN rank.